### PR TITLE
JWT and RPC Abstractions

### PR DESCRIPTION
**Pull request by TimGraf**

### CODE REVIEW

**Comment by TimGraf** [comment]
@AlexMaskovyak @gsiegman @atomboulian @arizonatribe @mgonzo @RegoAttainia @JonoMercier added as reviewers.

File: attainia_django_extensions/rpc/rpc_decorator.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+from nameko.rpc import rpc
+
+from cid import locals
+
+
+def rpc_decorator(function):
+    """ Wrap a Namkeo RPC call """
+
+    @rpc
+    def wrapper(self, cid, *args):
```
Comment (TimGraf) [inline attainia_django_extensions/rpc/rpc_decorator.py]: I think I'll need to add `**kwargs` as well.

File: attainia_django_extensions/rpc/rpc_mixin.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    # Nameko event dispatcher
+    dispatch = EventDispatcher()
+
+    def _get_connection_pool(self):
+        """ Get a connection from the pool """
+        # Connection pool for RPC cluster.
+        conn_pool_provider_path = import_string(settings.RPC_CONNECTION_POOL_PROVIDER)
+        return conn_pool_provider_path()
+
+    def call_service_method(self, service_name: str, method_name: str, use_async: bool, *args):
```
Comment (TimGraf) [inline attainia_django_extensions/rpc/rpc_mixin.py]: I think I'll need to add `**kwargs` as well.

File: attainia_django_extensions/rpc/rpc_decorator.py (hunk 1 above)
Comment (AlexMaskovyak) [reply attainia_django_extensions/rpc/rpc_decorator.py]: Yeah, and then to pass that onto the decorated function.

Comment (TimGraf) [reply attainia_django_extensions/rpc/rpc_decorator.py]: Yes, definitely.  Working on that now.

**Comment by RegoAttainia** [comment]
looks good.

What decision have we come to for the actual repo to use for this project?

**Comment by TimGraf** [comment]
I was just talking to @gsiegman about that.  I'm good with either one.  So far this is all very much none proprietary.

**Comment by RegoAttainia** [comment]
yeah and the changes that I made were also not proprietary.  I'm just wondering if we will down the road need make a proprietary version but we can cross that when needed.  We just need a decision.  I'm all for keeping it in here, we just need to concern ourselves with the public nature of the review conversations?

**Comment by AlexMaskovyak** [comment]
Yes, keep this in the public repo.

**Comment by gsiegman** [comment]
LGTM

**Comment by TimGraf** [comment]
@AlexMaskovyak is this OK to merge?

**Pull request merged**